lib/hanami/action.rb

```diff
@@ -28,44 +32,643 @@ module Hanami
+    }.freeze
+
+    # The request method
+    #
+    # @since 0.3.2
+    # @api private
+    REQUEST_METHOD = 'REQUEST_METHOD'.freeze
+
+    # The Content-Length HTTP header
+    #
+    # @since 1.0.0
+    # @api private
+    CONTENT_LENGTH = 'Content-Length'.freeze
+
+    # The non-standard HTTP header to pass the control over when a resource
+    # cannot be found by the current endpoint
+    #
+    # @since 1.0.0
+    # @api private
+    X_CASCADE = 'X-Cascade'.freeze
+
+    # HEAD request
+    #
+    # @since 0.3.2
+    # @api private
+    HEAD = 'HEAD'.freeze
+
+    # The key that returns router parsed body from the Rack env
+    ROUTER_PARSED_BODY = 'router.parsed_body'.freeze
+
+    # The key that returns accepted mime types from the Rack env
+    #
+    # @since 0.1.0
+    # @api private
+    HTTP_ACCEPT          = 'HTTP_ACCEPT'.freeze
+
+    # The header key to set the mime type of the response
+    #
+    # @since 0.1.0
+    # @api private
+    CONTENT_TYPE         = 'Content-Type'.freeze
+
+    # The default mime type for an incoming HTTP request
+    #
+    # @since 0.1.0
+    # @api private
+    DEFAULT_ACCEPT       = '*/*'.freeze
+
+    # The default mime type that is returned in the response
+    #
+    # @since 0.1.0
+    # @api private
+    DEFAULT_CONTENT_TYPE = 'application/octet-stream'.freeze
+
+    # @since 0.2.0
+    # @api private
+    RACK_ERRORS = 'rack.errors'.freeze
+
+    # This isn't part of Rack SPEC
+    #
+    # Exception notifiers use <tt>rack.exception</tt> instead of
+    # <tt>rack.errors</tt>, so we need to support it.
+    #
+    # @since 0.5.0
+    # @api private
+    #
+    # @see Hanami::Action::Throwable::RACK_ERRORS
+    # @see http://www.rubydoc.info/github/rack/rack/file/SPEC#The_Error_Stream
+    # @see https://github.com/hanami/controller/issues/133
+    RACK_EXCEPTION = 'rack.exception'.freeze
+
+    # The HTTP header for redirects
+    #
+    # @since 0.2.0
+    # @api private
+    LOCATION = 'Location'.freeze
+
     # Override Ruby's hook for modules.
     # It includes basic Hanami::Action modules to the given class.
     #
     # @param base [Class] the target action
     #
     # @since 0.1.0
     # @api private
-    #
-    # @see http://www.ruby-doc.org/core-2.1.2/Module.html#method-i-included
-    #
-    # @see Hanami::Action::Rack
-    # @see Hanami::Action::Mime
-    # @see Hanami::Action::Http
-    # @see Hanami::Action::Redirect
-    # @see Hanami::Action::Exposable
-    # @see Hanami::Action::Throwable
-    # @see Hanami::Action::Callbacks
-    # @see Hanami::Action::Validatable
-    # @see Hanami::Action::Configurable
-    # @see Hanami::Action::Callable
-    def self.included(base)
+    def self.inherited(base)
       base.class_eval do
-        include Rack
-        include Mime
-        include Redirect
-        include Exposable
-        include Throwable
-        include Callbacks
+        include Utils::ClassAttribute
+        class_attribute :before_callbacks
```

Comment:
Quick question: callbacks are included for making easier to share behaviour?

Reply:
yes

Reply:
Ok, I can live with that. 

Callbacks always make me antsy. I'd rather call `super` (or some other collaborator) on `#call`